ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,389 @@
+## Contents
+* [Custom NGINX configuration](#custom-nginx-configuration)
```

Comment:
# Customizing nginx

there are 3 ways to customize nginx
1. annoations: annotate the ingress, use this if you want to expose a few site specific features
2. config map: create a stand alone config map, use this if you want a lot of completey different features (reword)
3. custom template: when do people want this? 

## Annotations

The following annotaitons are supported:
table of annotations -> links to sections below

Authentication
Rewrite 
...


## Config Map

## Custom Template